src/Project.hs

```diff
@@ -74,21 +74,25 @@ data BuildStatus
   | BuildFailed (Maybe Text)
   deriving (Eq, Show, Generic)
 
--- When attempting to integrated changes, there can be three states: no attempt
--- has been made to integrate; integration (e.g. merge or rebase) was successful
--- and the new commit has the given sha; and an attempt to integrate was made,
--- but it resulted in merge conflicts.
+-- When attempting to integrated changes, there can be four states: no attempt
+-- has been made to integrate; unablet to integrate due to base branch not being master;
+-- integration (e.g. merge or rebase) was successful
+-- and the new commit has the given sha; an attempt to integrate was made,
+-- but it resulted in merge conflicts; and the integration is forbidden due to the target branch
+-- not being valid
 data IntegrationStatus
   = NotIntegrated
   | Integrated Sha BuildStatus
   | Conflicted Branch
+  | Forbidden
```

Comment:
I'm not sure `Forbidden` is the right term here. Forbidden can be for any reason. In this case it is because the base branch is not set, but one can also imagine the Forbidden state meaning there are orphan fixup commits (see #37). Would `IncorrectBaseBranch` be a better name?